android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java

```diff
@@ -12,17 +12,22 @@
 import com.appcoins.sdk.billing.ServiceConnectionException;
 import com.appcoins.sdk.billing.SkuDetailsResult;
 import com.appcoins.sdk.billing.service.WalletBillingService;
+
+import org.json.JSONException;
+
 import java.util.List;
 
 class AppCoinsAndroidBillingRepository implements Repository, ConnectionLifeCycle {
   private final int apiVersion;
   private final String packageName;
   private WalletBillingService service;
   private boolean isServiceReady;
+  private EventLogger eventLogger;
 
   public AppCoinsAndroidBillingRepository(int apiVersion, String packageName) {
     this.apiVersion = apiVersion;
     this.packageName = packageName;
+    this.eventLogger = new EventLogger();
```

Comment:
Maybe create the EventLogger in CatapultBillingAppCoinsFactory and pass by parameter. We create all the Billing dependencies in CatapultBillingAppCoinsFactory.

Reply:
Awesome point!